eth/stagedsync/stage_headers_snapshot.go

```diff
@@ -24,77 +25,94 @@ func StageHeadersSnapshotGenCfg(db ethdb.RwKV, snapshotDir string) HeadersSnapsh
 }
 
 func SpawnHeadersSnapshotGenerationStage(s *StageState, tx ethdb.RwTx, cfg HeadersSnapshotGenCfg, sm *snapshotsync.SnapshotMigrator, torrentClient *snapshotsync.Client, quit <-chan struct{}) error {
-	useExternalTx := tx != nil
-	if !useExternalTx {
-		var err error
-		tx, err = cfg.db.BeginRw(context.Background())
-		if err != nil {
-			return err
-		}
-		defer tx.Rollback()
+	//generate snapshot only on initial mode
```

Comment:
Detecting initial cycle by checking `tx == nil` is fragile. It is better to pass the argument `initialCycle` explicitly, like it is done in the `Headers` stage

Reply:
fixed